src/cpp/shared/adventure/adv.h

```diff
@@ -66,34 +67,30 @@ class armyGroup {
 
 class hero {
 public:
-  __int16 spellpoints;
+  short spellpoints;
   char idx;
   char ownerIdx;
-  char field_4;
-  char field_5;
-  char field_6;
-  char field_7;
-  char field_8;
-  char field_9;
-  char name[13];
+  short aiLastHeroInteractionTurn;
+  char aiLastHeroInteractionIdx;
+  short aiLastTownInteractionTurn;
+  char aiLastTownInteractionIdx;
+  char name[MAX_FILENAME_LENGTH];
```

Comment:
Should use a different constant. Although the max hero name is the same as the max filename length, hero names are not filenames, and they could be changed to differ.

We actually teach why some things should be duplicated in my software design business. See https://www.pathsensitive.com/2018/01/the-design-of-software-is-thing-apart.html

We actually collect examples of cases where duplication is good. I'm adding this there.

https://docs.google.com/document/d/1DvYcmeuadYEHY9uw8ZWpdXmem_pykgJUDDaCkd_VGJs/edit#

Reply:
I agree. Accidentally missed this